mythril/laser/plugin/plugins/plugin_annotations.py

```diff
@@ -41,29 +47,79 @@ def __copy__(self):
         return result
 
     def get_storage_write_cache(self, iteration: int):
-        if iteration not in self.storage_written:
-            self.storage_written[iteration] = []
-
-        return self.storage_written[iteration]
+        return self.storage_written.get(iteration, set())
 
     def extend_storage_write_cache(self, iteration: int, value: object):
-        if iteration not in self.storage_written:
-            self.storage_written[iteration] = [value]
-        elif value not in self.storage_written[iteration]:
-            self.storage_written[iteration].append(value)
-
-
-class WSDependencyAnnotation(StateAnnotation):
+        self.storage_written.get(iteration, set()).add(value)
+
+    def check_merge_annotation(self, other: "DependencyAnnotation"):
+        assert isinstance(other, DependencyAnnotation)
```

Comment:
```suggestion
        if not isinstance(other, DependencyAnnotation):
            raise TypeError("Expected instance of DependencyAnnotation")
```
I think this should raise a type error https://docs.python.org/3/library/exceptions.html#TypeError